invenio/modules/knowledge/models.py

```diff
@@ -166,6 +185,62 @@ def set_dyn_config(self, field, expression, collection=None):
                                     search_expression=expression,
                                     collection=collection)
 
+    @staticmethod
+    def generate_slug(name, tries=10):
```

Comment:
@jirikuncar I know that it's not the best way to generate slug... but it works if e.g. we already have in db two knowledge like: "FAQ" and "FAQ!".
And if someone try to add e.g. "faq" or "$FAQ", in this case we are sure that function returns a valid slug (not considering any racing conditions.. but this is another problem.. XD)